src/ServiceManagement/Profile/Commands.Profile/Account/AddAzureAccount.cs

```diff
@@ -18,57 +18,125 @@
         public PSCredential Credential { get; set; }
 
-        [Parameter(ParameterSetName = "ServicePrincipal", Mandatory = true)]
+        [Parameter(ParameterSetName = ServicePrincipalParameterSetName, Mandatory = false, HelpMessage = "Certificate thumprint, requires -ClientId and cannot be used with -Credential")]
+        [ValidateNotNullOrEmpty]
+        public string CertificateThumprint { get; set; }
+
+        [Parameter(ParameterSetName = ServicePrincipalParameterSetName, Mandatory = false, HelpMessage = "Application Id/Client Id, requires -CertificateThumbprint and cannot be used with -Credential")]
+        [ValidateNotNullOrEmpty]
+        public string ClientId { get; set; }
+
+        [Parameter(ParameterSetName = ServicePrincipalParameterSetName, Mandatory = true)]
         public SwitchParameter ServicePrincipal { get; set; }
         
-       [Parameter(ParameterSetName = "User", Mandatory = false, HelpMessage = "Optional tenant name or ID")]
-       [Parameter(ParameterSetName = "ServicePrincipal", Mandatory = true, HelpMessage = "Tenant name or ID")]
-       [Alias("TenantId")]
+        [Parameter(ParameterSetName = UserParameterSetName, Mandatory = false, HelpMessage = "Optional tenant name or ID")]
+        [Parameter(ParameterSetName = ServicePrincipalParameterSetName, Mandatory = true, HelpMessage = "Tenant name or ID")]
+        [ValidateNotNullOrEmpty]
+        [Alias("TenantId")]
         public string Tenant { get; set; }
-     
+
+        [Parameter(ParameterSetName = ServicePrincipalParameterSetName, Mandatory = false, HelpMessage = "Subscription Id")]
+        [ValidateNotNullOrEmpty]
+        public string SubscriptionId { get; set; }
+
         public AddAzureAccount() : base(true)
         {
             Environment = EnvironmentName.AzureCloud;
         }
 
         public override void ExecuteCmdlet()
         {
+            // Validate
+            if(ParameterSetName.Equals(ServicePrincipalParameterSetName))
+            {
+                if (string.IsNullOrEmpty(SubscriptionId))
+                {
+                    if (Profile.DefaultSubscription == null)
+                    {
+                        WriteErrorWithTimestamp("No Subscription Set. Please pass the -SubscriptionId/-SubscriptionName to set the default subscription.");
+                        return;
+                    }
+                    SubscriptionId = Profile.DefaultSubscription.Id;
+                }
+            }
+
+            if (Credential != null)
+            {
+                if (CertificateThumprint != null || ClientId != null)
+                {
+                    WriteErrorWithTimestamp("Credential cannot be used with CerficateThumprint and ClientId.");
+                    return;
+                }
+            }
+            else
+            {
+                if (CertificateThumprint == null || ClientId == null)
```

Comment:
I think you could consider moving the parameter input validation part to a separate function. 

BTW, the proper parameter sets should generally be enough to avoid the wrong combinations of user inputs.